Project-Aurora/Project-Aurora/Devices/UnifiedHID/UnifiedHID.cs

```diff
@@ -500,4 +562,460 @@ public bool SetBottomLed(byte r, byte g, byte b)
+                            0x6f ,0x6f ,0x00 ,0x6f ,0x65 ,0x65 ,0x66 ,0x5a ,0x65 ,0xea ,0xea ,0xdc ,0xea ,0xea ,0x00 ,0xea,
+                            0xdc ,0xdc ,0x00 ,0xce ,0xdc ,0xea ,0xe7 ,0xe5 ,0xe7 ,0xe5 ,0xe5 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00,
+                            0x00 ,0x65 ,0x5a ,0x50 ,0x5a ,0x50 ,0x50 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0xdc ,0xce ,0xc0,
+                            0xce ,0xc0 ,0xc0 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0xe7 ,0x00 ,0x00 ,0xe2 ,0xe2 ,0xe2 ,0xe2,
+                            0xdf ,0xdf ,0xdf ,0xdf ,0xdf ,0x5a ,0x00 ,0x00 ,0x45 ,0x45 ,0x45 ,0x45 ,0x3b ,0x3b ,0x3b ,0x3b,
+                            0x3b ,0xce ,0x00 ,0x00 ,0xb2 ,0xb2 ,0xb2 ,0xb2 ,0xa4 ,0xa4 ,0xa4 ,0xa4 ,0xa4 ,0xdc ,0xdc ,0xdc,
+                            0xdc ,0x00 ,0xda ,0xda ,0xda ,0xda ,0xda ,0x00 ,0xd7 ,0x30 ,0x30 ,0x30 ,0x30 ,0x00 ,0x26 ,0x26,
+                            0x26 ,0x26 ,0x26 ,0x00 ,0x1c ,0x96 ,0x96 ,0x96 ,0x96 ,0x00 ,0x88 ,0x88 ,0x88 ,0x88 ,0x88 ,0x00,
+                            0x7a ,0xd7 ,0xd7 ,0xd7 ,0x00 ,0xd4 ,0xd4 ,0xd4 ,0xd4 ,0xd4 ,0xd1 ,0xd1 ,0xd1 ,0x1c ,0x1c ,0x1c,
+                            0x00 ,0x11 ,0x11 ,0x11 ,0x11 ,0x11 ,0x06 ,0x06 ,0x06 ,0x7a ,0x7a ,0x7a ,0x00 ,0x6c ,0x6c ,0x6c,
+                            0x6c ,0x6c ,0x5e ,0x5e ,0x5e ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00,
+                            0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00,
+                            0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x00 ,0x24 ,0xcf};
+
+                    break;
+                case 0x13:
+                    buffer = new byte[] { 0x13, 0x08, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00 };
+                    //length = 8;
+                    break;
+            }
+
+
+            if (buffer == null)
+            {
+                return false;
+            }
+
+
+            bool success = ctrl_device.WriteFeatureData(buffer);
+
+            return success;
+        }
+
+        public RoccatVulcan()
+        {
+            IsKeyboard = true;
+        }
+
+        public override bool Connect()
+        {
+            if (!Global.Configuration.VarRegistry.GetVariable<bool>($"UnifiedHID_{this.GetType().Name}_enable"))
+            {
+                return false;
+            }
+
+            IEnumerable<HidDevice> devices = HidDevices.Enumerate(0x1E7D, new int[] { 0x307A, 0x3098 });
+
+            try
+            {
+                if (devices.Count() > 0)
+                {
+                    HidDevice[] devicearray = devices.ToArray();
+                    ctrl_device_leds = devices.FirstOrDefault(dev => dev.Capabilities.UsagePage == 0x0001 && dev.Capabilities.Usage == 0x0000);
+                    ctrl_device = devices.FirstOrDefault(dev => dev.Capabilities.FeatureReportByteLength > 50);
```

Comment:
Should this really be `FirstOrDefault`? Shouldn't it not continue if it cannot find it?